### ISSUE TITLE
loss never goes down

### ISSUE THREAD

**Issue by jasondesante** [tensorflow/magenta#891]
I am having a lot of issues training models.  I am following all the instructions but there simply isn't enough information to explain to someone what to do.  

I am training performance_rnn.  I am training with a larger dataset (26mb of midi = 74mb notestream = 8gb performance_rnn training data).  Training loss has been at 5 since I started training 12 hours ago.  This is clearly a problem.

The last thing I trained resulted in the eval loss going UP from the very beginning.   I was told that I didn't use a large enough dataset because it must have been overfitting.  I figured that would be a lot larger dataset because the last one was only 1 gb of performance_rnn training data, and
This time after 12 hours of training, the training set is still at loss 5.....basically the exact same spot it was at when it started.......

I have to be missing something here because I feel like I'm following all the instructions and all I get is problems and things breaking. Every single time I've done this the loss has gone down, except for now.

If everything I have tried has basically failed I think it's fair to say that you guys need to explain yourselves better because I have been trial and erroring this thing for days and the best I got is a model trained on Steely Dan that knows how to do add 9 chords really well (which is nice because that's the trademark Steely Dan chord, but still.....I feel like things could work better).  

You guys need to be more specific or at the very least not close issues that are definitely issues that need solving.  I have been slowly training larger and larger datasets with no success and many problems that are completely different.  I can't see a relationship between eval loss going up at one time, and training loss not moving at all another time.  I can't understand why those things are happening all I know is that from my previous experience that hadn't happened before.  

I spent a couple days just making notestreams trying to figure out why sometimes they make datasets of 0 bytes and I still don't understand why and haven't found anyone saying anywhere any information about that issue.  I thought it would be so common that someone in the Magenta team would have bothered to post their experiences with training certain sized datasets and what to expect so others don't need to waste dozens of hours repeating the same stuff.  I will post my findings on that later but right now I can't get a model to train.

Please help I don't feel like opening a new issue every time someone closes one I make because they assume I'm just being stupid and it's easy to fix.  I'm not being stupid I'm following all your instructions, and getting fairly frustrated by it because I'm not doing anything creative.

**Comment by jasondesante** [comment]
its at 1800 steps and loss is still at 5....I started up eval when I got up and it seems that eval is also going up.  Do I need to wait until 20,000 steps until I can judge it?  Is this now too big of a dataset?  Too large too small, what is just right?  I have literally no clue what to do next.



I think maybe its just the midi files and the model wants midi files formatted a certain way and if I prepare the midis the way it wants maybe it will perform properly.  That's the only thing I can think about, but there's no way of knowing because all I'm doing is running these terminal scripts, and not bringing me closer to any sort of understanding.  

Please help me I actually want to get into this stuff but it's almost impossible with the lack of information, lack of people discussing these common things in forums.  Even though I want to put all my time into learning this is it just too much?  Only for the people who already know what they're doing and I should just go away?

**Comment by jasondesante** [comment]
another thing is the loss numbers in tensorboard say one thing and in actual terminal, when you look at the numbers in the eval window, it ALWAYS shows perplexity = 0 loss = 0 accuracy = 0.  Every single time I've ever run it, it has showed 0 for everything in terminal.  In tensorboard the numbers are always numbers and not 0.  That's definitely some sort of bug!

**Comment by Calvin-Shen** [comment]
I just facing the same problem like you~, do you have solution yet? @jasondesante

**Comment by Calvin-Shen** [comment]
can you run train script and train --eval script at the same time?  I can't do this, otherwise it will make a lot of errors~

**Comment by jasondesante** [comment]
@Calvin-Shen  Sorry I haven't found a solution yet.  I have spent less time playing with this stuff now that I've realized it isn't as inviting of a community as I once thought.  The criticism I've seen from people that actually know what they're doing is that Magenta doesn't have enough documentation, and I understand what they're saying because I am clueless.

I've realized Magenta is only for people that already have years of experience and know a lot of what to do.  If you are learning through reading websites and courses because of this.....it seems like you will still get no help here. 

I watched a lot of videos from a guy called "Doug Eck" who apparently works on Magenta....and he would say that they need more musicians and normal people popping in to give feedback and get involved, and that the people on the google brain team are always in the forums answering questions.  I don't know if that's a lie or just his intentions that didn't come true but I feel disappointed by the disconnect between the intentions and the reality of this project.

I simply want to learn!  If I'm here alone in the forums just venting my feelings to myself it is a very discouraging thing because I know how much farther along I would be if someone had just shared some simple information.

TLDR, No and I'm frustrated

**Comment by Calvin-Shen** [comment]
@jasondesante yesterday,I run the performance_rnn_train script, and even 20000 iteration.The loss was always keeping about 4.0 or so,and the accuracy was about 0.05 or so.However to my happiness when I saw it again at 4.00 p.m,the loss has dropped about 0.05 and the accuracy was about 97%,and the iteration counts was about 75000 at that time,it takes me nearly a whole day to do that.
maybe the instruction is not wrong.
Hope my experience can help you~ best wishes!!!

**Comment by jasondesante** [comment]
@Calvin  Hey dude thanks for the communication!  That's literally all we need to learn!  Google should hire you now that you've proven to be more dedicated to building a community over here than the guy who said his job is to build a community over here. (Doug Eck please be more to your word and actually come up on here and build a community instead of leaving the people alone to ask questions to a void which then prevents the possibility of a community ever appearing).
